src/vm/mod.rs

```diff
@@ -2,9 +2,582 @@
 //! Note that these modules are public for documentation visiblility,
 //! But should never be used outside of the module by `common` or `compiler`.
 
-pub mod vm;
```

Comment:
I'll move this file for now, but it the future it might be better to name the module something else, like `interpreter`.